Refactor test_cron test

waynew opened the pull request:
### What does this PR do?

Refactors the one `test_cron` state test that we have, because #52079

It's possible that the dict could contain some more helpful information
when/if this test fails again. This assertion improves the potential for
getting useful information in the test failure.

### What issues does this PR fix or reference?

#52079

### Previous Behavior

The assert statement failed to have any useful information in it.

### New Behavior

Now when the test fails, it should have some useful information in the failure

### Tests written?

Yes (changed the test - verified that changing the expected value failed the test and provided useful information, i.e. the entire contents of the changes, then changed it back and verified that the test passes)

### Commits signed with GPG?

Yes

waynew committed:
Refactor test_cron test

It's possible that the dict could contain some more helpful information
when/if this test fails again. This assertion improves the potential for
getting useful information in the test failure.

dwoz committed:
Merge branch '2019.2' into 52079-refactor-test

dwoz reviewed (changes requested):
@waynew Please have a look at the test failures.

waynew committed:
Add attrs for Ubuntu

For some reason an empty attrs shows up in changes. This is will go
ahead and add attrs to the expected changes if it exists.

waynew commented:
Well, I fixed the failing Ubuntu tests @dwoz - it just completely failed running on win 2016. I'm re-running that job 🤞 (but in general I don't think that that failure is related to my changes)

waynew commented:
Currently still running, so that's a promising sign

Ch3LL committed:
Merge branch '2019.2' into 52079-refactor-test
